### PR TITLE
Provide mechanism for configuring submodule aliases

### LINKED ISSUE #15178: Wrong link to submodule in WebUI
- Gitea version (or commit ref): 1.13.6
- Git version: 2.20.1
- Operating system: 
```
$ uname -amrs
Linux control01 5.10.17-v7+ #1403 SMP Mon Feb 22 11:29:51 GMT 2021 armv7l GNU/Linux
```
```
$ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
```

- used gitea package: [gitea-1.13.6-linux-arm-6](https://dl.gitea.io/gitea/1.13.6/gitea-1.13.6-linux-arm-6)
  
- gitea usage: systemd
  
- distribution: Raspbian 10
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
- Log gist:

## Description

Hi, I updated from version 1.12.6 to 1.13.6 and have the following problem now.
My gitea is configured for HTTPS and my submodule is added via SSH.
From the main repo the link to the submodule is only HTTP so I can't access
my submodule from the web UI. In version 1.12.6 this worked as expected.

See below some informations:
```
$ git config --file=.gitmodules -l
submodule.terraform.path=terraform
submodule.terraform.url=gitea@172.16.4.20:steled/terraform.git
submodule.terraform.update=rebase

```
The web gui is showing http://172.16.4.20/steled/terraform but should show https://172.16.4.20/steled/terraform

```
SSH_DOMAIN = %(DOMAIN)s
PROTOCOL = https
DOMAIN = ***.duckdns.org
HTTP_PORT = 443
ROOT_URL = %(PROTOCOL)s://%(DOMAIN)s:%(HTTP_PORT)s/
```

When I manually set SSH_DOMAIN to 172.16.4.20 the WebUI shows https://***.duckdns.org/steled/terraform and the clone link shows gitea@172.16.4.20:steled/terraform.git

Would be great if you can have a look at.

Thanks and best regards

## Screenshots

gitea_submodule_link

### PR DESCRIPTION
**Pull request by zeripath**
Sometimes Gitea users would like submodules when displayed on the website to have
a different alias than what is normally inferred by gitea.

This PR provides a mechanism to configure this.

Fix #15178

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by lunny** [comment]
Or we could have a configuration on repository setting UI.

**Comment by zeripath** [comment]
> Or we could have a configuration on repository setting UI.

Yes, were would need a common repository settings table:

```go
type Setting struct {
  ID int64
  UserID int64
  RepoID int64
  Key string
  Value string
}
```

Which could be used to store per user/per repo settings, with Key, RepoID and UserID forming an unique index.

Then we can remove some crap from repo and user, and allow other common settings

**Review by techknowlogick — APPROVED** [review]

**Comment by 6543** [comment]
please resolve conflicts

**Comment by zeripath** [comment]
conflicts resolved @6543

File: custom/conf/app.example.ini [hunk 1]
```diff
@@ -573,7 +573,17 @@ PATH =
...
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; [git.submodule]
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Submodule aliase
+; MAP_NAME_1 = git@127.0.0.1
+; MAP_VALUE_1 = https://example.com
+; MAP_NAME_2 = git@github.com/go-gitea/gitea
+; MAP_VALUE_2 = https://gitea.com/gitea/gitea
```
Comment (a1012112796) [inline custom/conf/app.example.ini:590]:
```suggestion
; MAP_VALUE_2 = https://gitea.com/gitea/gitea
; ...
```

**Comment by zeripath** [comment]
conflicts fixed again

**Pull request closed without merging**